### PR TITLE
[SCSS | Sass] Allow hyphens after parent selector

### PR DESCRIPTION
**Pull request by bgriffith**
This PR fixes the same issue as was fixed for classes in https://github.com/tonyganch/gonzales-pe/pull/179, but this time for parent selectors.

The fix is a duplicate of the solution applied to `checkClass`/`getClass` in #179.

The following used to cause an issue but is now fixed in this PR.

``` sass
&-#{$s1} {
  color: red;
}
```

### CODE REVIEW

File: src/sass/parse.js [hunk 1]
```diff
@@ -3239,55 +3239,80 @@ function checkParentSelector(i) {
...
-
-  let start = i;
+  const start = i;
   let l;
 
+  if (i >= tokensLength) return 0;
+
   while (i < tokensLength) {
     if (l = checkNumber(i) || checkIdentOrInterpolation(i)) i += l;
+    else if (tokens[i].type === TokenType.HyphenMinus) i += 1;
```
Comment (tonyganch) [inline src/sass/parse.js:3266]: So, `&----` is a valid selector, right?

File: src/sass/parse.js [hunk 2]
```diff
@@ -3239,55 +3239,80 @@ function checkParentSelector(i) {
...
 
   while (pos < tokensLength) {
     if (checkNumber(pos)) {
       content.push(getNumber());
     } else if (checkIdentOrInterpolation(pos)) {
       content = content.concat(getIdentOrInterpolation());
-    } else {
-      break;
-    }
+    } else if (tokens[pos].type === TokenType.HyphenMinus) {
```
Comment (tonyganch) [inline src/sass/parse.js:3289]: Here you propose that in `&----` there will be four `ident` nodes with content `-` in each. Should not it be just one node with content `----`?

File: src/sass/parse.js (hunk 1 above)
Comment (bgriffith) [inline src/sass/parse.js:3266]: Haha yep!

File: src/sass/parse.js (hunk 2 above)
Comment (bgriffith) [inline src/sass/parse.js:3289]: When you have more than one `-` in a row it gets picked up by the `checkIdent` function above so never reaches that condition.

Line 3289 only caters for when there's a single `-` say after `&-#{$foo}`.

So `&----` actually only returns 1 ident node.

I'll add a test to make this clearer :+1:

**Comment by bgriffith** [comment]
@tonyganch Thanks for the feedback! Added some extra tests, hopefully makes it a bit clearer

File: src/sass/parse.js (hunk 2 above)
Comment (tonyganch) [inline src/sass/parse.js:3289]: That's magic!

**Comment by tonyganch** [comment]
Merged into dev 🐙

**Pull request closed without merging**